### PR TITLE
Add flags for explicit ipv4/ipv6 control

### PR DESCRIPTION
**Pull request by tiedotguy**
Happy eyeballs might be good for browsers, but at this volume it is better to be explicit than to guess.

### CODE REVIEW

File: pkg/backends/datadog/datadog.go [hunk 1]
```diff
@@ -258,28 +258,44 @@ func NewClientFromViper(v *viper.Viper) (gostatsd.Backend, error) {
...
+	ipv4 := dd.GetBool("ipv4")
+	ipv6 := dd.GetBool("ipv6")
+	var network string
+	if ipv4 && ipv6 {
+		network = "tcp"
+	} else if ipv4 {
+		network = "tcp4"
+	} else if ipv6 {
+		network = "tcp6"
+	}
```
Comment (ash2k) [inline pkg/backends/datadog/datadog.go]: Should probably return an error if both are false?

File: pkg/backends/datadog/datadog.go [hunk 2]
```diff
@@ -258,28 +258,44 @@ func NewClientFromViper(v *viper.Viper) (gostatsd.Backend, error) {
...
-func NewClient(apiEndpoint, apiKey string, metricsPerBatch uint, compressPayload, dualStack bool, clientTimeout, maxRequestElapsedTime time.Duration) (*Client, error) {
+func NewClient(apiEndpoint, apiKey, network string, metricsPerBatch uint, compressPayload bool, clientTimeout, maxRequestElapsedTime time.Duration) (*Client, error) {
 	if apiEndpoint == "" {
 		return nil, fmt.Errorf("[%s] apiEndpoint is required", BackendName)
 	}
 	if apiKey == "" {
 		return nil, fmt.Errorf("[%s] apiKey is required", BackendName)
 	}
+	if network == "" {
+		return nil, fmt.Errorf("[%s] at least one of ipv4 or ipv6 must be enabled", BackendName)
```
Comment (ash2k) [inline pkg/backends/datadog/datadog.go]: Huh, this is the error that should be returned from `NewClientFromViper()`.
And this function does not know anything about ipv4/ipv6 flags. It should error out with "network must not be empty". WDYT?

Comment (ash2k) [reply pkg/backends/datadog/datadog.go]: Not for this PR: we should remove backend name from error string and instead add it up the call stack.

File: pkg/backends/datadog/datadog.go (hunk 1 above)
Comment (tiedotguy) [reply pkg/backends/datadog/datadog.go]: I had that initially, then moved the validation logic to be with the rest of its friends below.

File: pkg/backends/datadog/datadog.go (hunk 2 above)
Comment (tiedotguy) [reply pkg/backends/datadog/datadog.go]: I think enforcing 4/6 at all is not our problem, and we can pass it off to anything Go supports :)

Comment (tiedotguy) [reply pkg/backends/datadog/datadog.go]: Regarding adding it up the callstack: it should probably be on the logger...  The one we add when we do #2

**Pull request merged**